tests/serialization_tests.rs

```diff
@@ -63,6 +63,90 @@ fn serialize_felts(felts: &[gnark_backend_wrapper::Fr]) -> Vec<u8> {
     buff
 }
 
+fn ping_pong(
+    encoded_element: String,
+    go_function: unsafe extern "C" fn(gnark_backend_wrapper::GoString) -> *const ffi::c_char,
+) -> &'static [u8] {
+    // Prepare ping for Go.
+    let pre_ping = ffi::CString::new(encoded_element).unwrap();
+    let ping = gnark_backend_wrapper::GoString::try_from(&pre_ping).unwrap();
+
+    // Send and receive pong from Go.
+    let pong: *const ffi::c_char = unsafe { go_function(ping) };
+
+    // Prepare pong for Rust.
+    let go_pre_serialized_element = unsafe { ffi::CStr::from_ptr(pong) };
+
+    go_pre_serialized_element.to_str().unwrap().as_bytes()
+}
+
+fn random_add_term() -> AddTerm {
+    // Sample random coefficient.
+    let coefficient: gnark_backend_wrapper::Fr = rand::random();
+    // Sample a random sum.
+    let sum = acvm::Witness::new(rand::random());
+    AddTerm { coefficient, sum }
+}
+
+fn random_mul_term() -> MulTerm {
+    // Sample random coefficient.
+    let coefficient: gnark_backend_wrapper::Fr = rand::random();
+    // Sample a random multiplicand.
+    let multiplicand = acvm::Witness::new(rand::random());
+    // Sample a random multiplier.
+    let multiplier: acvm::Witness = acvm::Witness::new(rand::random());
+
+    MulTerm {
+        coefficient,
+        multiplicand,
+        multiplier,
+    }
+}
+
+fn random_mul_terms(count: usize) -> Vec<MulTerm> {
+    (0..count).into_iter().map(|_| random_mul_term()).collect()
+}
+
+fn random_add_terms(count: usize) -> Vec<AddTerm> {
+    (0..count).into_iter().map(|_| random_add_term()).collect()
+}
+
+fn random_raw_gate(term_count: usize) -> RawGate {
+    let add_terms = random_add_terms(term_count);
+    let mul_terms = random_mul_terms(term_count);
+    let constant_term: gnark_backend_wrapper::Fr = rand::random();
+    RawGate {
+        mul_terms,
+        add_terms,
+        constant_term,
+    }
+}
+
+fn random_raw_gates(count: usize, term_count: usize) -> Vec<RawGate> {
+    (0..count)
+        .into_iter()
+        .map(|_| random_raw_gate(term_count))
+        .collect()
+}
```

Comment:
I'm not sure about this one 
```suggestion
fn random_raw_gates(count: usize, term_count: usize) -> Vec<RawGate> {
    vec![random_raw_gate(term_count); count]
}
```